### PR TITLE
Update CITATION.cff to fix Zenodo, add CFF validator

### LINKED ISSUE #407: Fix Zenodo and add CFF validator pre-commit hook
https://github.com/citation-file-format/cffconvert

Zenodo DOI issuance has been failing since 0.6.1. Let's use a validator to help us find the issue.

### PR DESCRIPTION
**Pull request by MattF-NSIDC**
Currently, Zenodo is unhappy with array values for `license`. This violates the CFF spec. The related GitHub issue is added as a comment in this PR.

Resolves #407 

This has been (silently) going on for about a month, so our last DOI'd release is 0.6.1 :sob: I added the new pre-commit check to hopefully avoid this going forward, but for this particular issue, it wouldn't have helped, since it implements the spec more accurately than Zenodo (for this particular field at least).

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -4,15 +4,18 @@ message: |
...
   - name: "The earthaccess community"
     website: "https://github.com/nsidc/earthaccess/discussions"
   - name: "NSIDC"
     email: "nsidc@nsidc.org"
 type: "software"
-license: ["MIT"]
+# NOTE: The CFF spec says `license` can be a list, but Zenodo is currently not
+#       accepting lists for this key:
+#       https://github.com/zenodo/zenodo/issues/2515
+license: "MIT"
```
Comment (MattF-NSIDC) [inline CITATION.cff:18]: This was the thing that broke everything. All the other changes are from adding the validator.

File: CITATION.cff [hunk 2]
```diff
@@ -27,51 +30,57 @@ version: "0.8.2"
...
+  - type: "grant"
+    title: "Openscapes: Supporting better science for future us"
+    institution:
+      name: "National Aeronautics and Space Administration"
+    number: "20-TWSC20-2-0003"
+    authors:
+      - family-names: "Lowndes"
+        given-names: "Julia"
+        orcid: "https://orcid.org/0000-0003-1682-3872"
+        website: "https://github.com/jules32"
```
Comment (MattF-NSIDC) [inline CITATION.cff:86]: @jules32 `title` and `authors` are required fields now, and I took a shot, not my best shot, but a compromise between goodness/fastness :) Did I get the title/authors of the grant correct?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -22,3 +22,9 @@ repos:
     hooks:
       - id: prettier
         types_or: [yaml]
+  - repo: https://github.com/citation-file-format/cffconvert
+    rev: "054bda51dbe278b3e86f27c890e3f3ac877d616c"
+    hooks:
+      - id: "validate-cff"
+        args:
+          - "--verbose"
```
[the remark this answers is not in the document]
Comment (MattF-NSIDC) [reply .pre-commit-config.yaml:30]: I just thought of one. We could have the prettier hook run twice; once for the normal stuff we want to format, and again ONLY for CITATION.cff!

```yaml
  - repo: "https://github.com/pre-commit/mirrors-prettier"
    rev: "v3.1.0"
    hooks:
      - id: "prettier"
        files: "CITATION.cff"
```

I'd need to test it. But I don't have time right now.

File: CITATION.cff (hunk 2 above)
Comment (jules32) [reply CITATION.cff:86]: Hi @MattF-NSIDC ! Awesome! Please also add: 

```
      - family-names: "Robinson"
        given-names: "Erin"
        orcid: "https://orcid.org/0000-0001-9998-0114"
        website: "https://erinrobinson.info/"
```

Comment (MattF-NSIDC) [reply CITATION.cff:86]: :heavy_check_mark: d9994e1

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -22,3 +22,9 @@ repos:
     hooks:
       - id: prettier
         types_or: [yaml]
+  - repo: https://github.com/citation-file-format/cffconvert
+    rev: "054bda51dbe278b3e86f27c890e3f3ac877d616c"
```
Comment (MattF-NSIDC) [inline .pre-commit-config.yaml:26]: I don't like this either, but it's what they documented. Opened issue: https://github.com/citation-file-format/cffconvert/issues/375

**Comment by MattF-NSIDC** [comment]
If anyone approves, this is ready to merge on my end, so go for it :)

**Review by betolink — APPROVED** [review]
Thanks for fixing it Matt!

**Pull request merged**